### DISCUSSION TITLE
Theming/UI preferences

### DISCUSSION THREAD

**Discussion by lolbat** [notaapp/talk#239] [Feature Requests]
Any chance that there will be an option for people to theme Nota or to use the standard OS UI?

I really do like the app but I am not really a fan of the UI and even just being able to default back to the system UI would be great.

**Comment by astoilkov** [comment]
We are considering theming. However, we will start with supporting changing base colors and hiding elements. I suppose you want to have full control over the UI?

BTW, which app UI do you like? Maybe we can take inspiration from there. Thanks.

**Reply by lolbat** [reply]
>  I suppose you want to have full control over the UI?

Eventually but even more course control like setting sidebar colours and changing the window controls. The current graphic at the top of the sidebars could be replaced with something more basic or even just some colours.

> BTW, which app UI do you like? 

Obsidian

**Reply by webwesen** [reply]
i'd probably suggest starting with light/dark mode selection (not tied to OS setting) as a 'gateway drug' (joking)

**Reply by astoilkov** [reply]
Got it. Do you use a theme for Obsidian? Or is it something you have something custom-built?

**Reply by lolbat** [reply]
I use the Minimal theme. 

A good Mac app to look at for ideas is Nova by Panic

https://nova.app

**Reply by webwesen** [reply]
i usually install a Bear-inspired theme in all apps that support it (Logseq, for example)

**Comment by exceptionraiser** [comment]
I'm wanting to style individual elements like tags, wiki links, todos, etc.

**Comment by vladstudio** [comment]
To be honest, theming is a dangerous road. The current look&feel of Nota is excellent as it is, and giving too much freedom to customize it might hurt in long term.

Sure, there are many excellent themes for VScode, and couple of great themes for Obsidian... but so many more are unfortunately not so great. 

I guess what i'm trying to say is - implementing theming in a "correct" way is much harder than it seems at first, and it is not obvious at first what the "correct" way is for any given app. A lot of analysis should be done.

**Reply by NoodleNodes** [reply]
Agreed. Nota looks beautiful as is. Maybe a couple of refinements to how tags are displayed, similar to iA Writer’s pill shaped tags, and possibly the addition of line spacing to make dense content easier to read. Otherwise very happy with the clean interface.

**Comment by marlonjames71** [comment]
I'd have to agree with the comment above about theming being a dangerous road. One of the things that attracted me to Obsidian at first was all the customization, but then that was the very thing that started frustrating me as it can be so distracting and time consuming.

I think Nota looks pretty great as is. I would tweak certain things like slightly more rounded corners on inline code & code blocks, the checkboxes could be nicer looking (white backgrounds in dark mode don't look so great), and stuff like that. Some nice polish around the app I think would be perfect.

**Reply by erusev** [reply]
> I would tweak certain things like slightly more rounded corners on inline code & code blocks, the checkboxes could be nicer looking (white backgrounds in dark mode don't look so great), and stuff like that. Some nice polish around the app I think would be perfect.

I like all of these! I'm adding them to my to-do list. If you have any other suggestions, I'd love to hear them

**Reply by marlonjames71** [reply]
Glad you like those ideas. I'll make sure to recommend more as they come up. 

One that I forgot is highlighting. Rounded corners would be great for this as well. And if the highlight color could be made brighter, that would be great as it's a bit hard to see right now. I'm a huge fan of Bear's styling in their new editor Panda. Posting some images for inspiration:

**Dark Mode** 
Nice, clear highlighting color. I think a nice bright peach color with dark text would look clean for Nota and it would match the syntax colors for your code blocks. Blue is always clean too though — only thing is Nota's selection color is already blue and I think it's fantastic the way it is currently, especially because of how the cursor is still very visible during selection.
<img width="694" alt="CleanShot 2022-08-06 at 19 35 37@2x" src="https://user-images.githubusercontent.com/48145861/183269099-0a32efb8-2b89-4c77-84b9-14941f1e5d4c.png">

**Light Mode**
I'm definitely a fan of this highlight color in light mode. So clean.
<img width="690" alt="CleanShot 2022-08-06 at 19 35 20@2x" src="https://user-images.githubusercontent.com/48145861/183269128-dbe20882-3f13-45f5-9d94-30e1572ef1bb.png">

I do like the simple and minimalistic style of the tags in Nota. But if you guys were to revamp those as well, I don't think you can go wrong with a nice pill shape background.

**Oh, and here are Panda's checkboxes (Dark & Light):**
<img width="244" alt="CleanShot 2022-08-06 at 19 50 02@2x" src="https://user-images.githubusercontent.com/48145861/183269343-2d3c3be2-c682-48a5-a092-746c60b97370.png">
<img width="240" alt="CleanShot 2022-08-06 at 19 51 20@2x" src="https://user-images.githubusercontent.com/48145861/183269370-8f3af748-9d7b-4132-bf11-ff49279e3794.png">

**Here are Craft's checkboxes (Dark & Light):**
<img width="237" alt="CleanShot 2022-08-06 at 19 48 52@2x" src="https://user-images.githubusercontent.com/48145861/183269315-3400e7ff-c8d2-4ecf-8920-bd6eadd5a263.png">
<img width="240" alt="CleanShot 2022-08-06 at 19 51 34@2x" src="https://user-images.githubusercontent.com/48145861/183269371-05607a61-a400-47ce-ad7c-1401c0ce0701.png">


Both are super clean looking. Anyway, thanks again for hearing me out!

**Reply by erusev** [reply]
Thanks so much for taking the time to share these ideas 🙏 I'll see if we can fit some of them in 0.39, which should come out in late August / early September ✨

**Reply by marlonjames71** [reply]
> Thanks so much for taking the time to share these ideas 🙏 I'll see if we can fit some of them in 0.39, which should come out in late August / early September ✨

Hey @erusev , I know you guys spent your time basically re-writing Nota which is awesome. I was just curious if you guys still planned on updating some of the UI as we spoke about above?

Thanks!

**Reply by erusev** [reply]
Absolutely, it should be easier now -- will try to address these for `0.40`.

**Reply by marlonjames71** [reply]
Very cool! Looking forward to it. Thanks for the reply.
